tlog-tiles.md

```diff
@@ -0,0 +1,157 @@
+`x001/x234/067`. (This allows storing tile resources efficiently in a filesystem
+without file/directory conflicts, and serving them directly.)
+
+The `.p/<W>` suffix is only present for partial tiles, defined below. `<W>` is
+the width of the tile, a decimal ASCII integer between 1 and 255, with no
+additional leading zeroes.
+
+This endpoint is immutable, so its caching headers SHOULD be long-lived.
+
+Full tiles MUST be exactly 256 hashes wide, or 8,192 bytes.  At “level 0” tiles
+contain leaf hashes. At “level 1” and above, each hash in a tile is the Merkle
+Tree Hash of a *full* tile at the level below.
+
+More formally, in the language of RFC 6962, Section 2.1, the *n*-th tile at
+level *l*, with *n* and *l* starting at 0, is the sequence of the following
+Merkle Tree Hashes, with *i* from 0 to 255:
+
+    MTH(D[(n * 256 + i) * 256**l : (n * 256 + i + 1) * 256**l])
+
+Note that a tile represents the entire subtree of height 8 with its hashes as
+the leaves. The Merkle Tree levels between those expressed by the tile hashes
+are reconstructed by hashing the leaves.
+
+#### Partial Tiles
+
+Some or all of the rightmost tiles in a tree will always be _partial_. A partial
+tile contains between 1 and 255 hashes, and MUST NOT be hashed into a tile at
+the level above.
+
+The partial tile at level *l* for a tree of size *s* has `floor(s / 256**l) mod
+256` entries. Empty tiles MUST NOT be served.
+
+For example, a tree of size 70,000 will be represented by 273 full level 0
+tiles, one partial level 0 tile of width 112, one full level 1 tile, one partial
+level 1 tile of width 17, and one partial level 2 tile of width 1. Note that a
+tree of size 256 will be represented by a full level 0 tile and a partial level
+1 tile of width 1.
+
+Logs MUST serve partial tiles corresponding to tree sizes for which a checkpoint
+was produced, but MAY delete any partial tile once the corresponding full tile
+becomes available. Clients MUST NOT fetch arbitrary partial tiles without
+verifying a checkpoint with a size that requires their existence, and MAY fetch
+the full tile in parallel as a fallback in case the partial tile is not
+available anymore.
+
+### Log entries
+
+The log entries are served as a “entry bundles” at
+
+	<prefix>/tile/entries/<N>[.p/<W>]
+
+with `Content-Type: application/data`.
+
+`<N>` and `.p/<W>` have the same meaning as in Merkle Tree tile paths above.
+
+Data tiles SHOULD be compressed at the HTTP layer. Logs MAY use
+`Content-Encoding: gzip` with no negotiation, or any compression algorithm
+requested by the client with `Accept-Encoding`. Clients SHOULD include `gzip`
+and `identity` in their `Accept-Encoding` headers.
+
+This endpoint is immutable, so its caching headers SHOULD be long-lived.
+
+Data tiles are sequences of big-endian uint16 length-prefixed entries. Each
+entry in a data tile hashes to the corresponding entry in the corresponding
+“level 0” tile according to RFC 6962, Section 2.1.
```

Comment:
I'm having a hard time finding "level 0" or "tile" mentioned in RFC 6962. Which part of Section 2.1 do you have in mind?

```The hash of a list with one entry (also known as a leaf hash) is: ... ```

Or perhaps
```The input to the Merkle Tree Hash is a list of data entries; these entries will be hashed to form the leaves of the Merkle Hash Tree.```
?

Reply:
This sentence needs reworking because I can completely see your confusion. What it is trying to say is that the hashing method is the same as RFC 6962, i.e. the value that is hashed has a domain separation prefix applied to it (https://github.com/transparency-dev/merkle/blob/main/rfc6962/rfc6962.go#L24)

Reply:
Indeed, split up the sentences to clarify.